qa/vector/build.gradle

```diff
@@ -51,13 +51,27 @@ tasks.register("checkVec", JavaExec) {
   systemProperty "es.logger.out", "console"
   systemProperty "es.logger.level", "INFO"  // Change to DEBUG if needed
   systemProperty 'es.nativelibs.path', TestUtil.getTestLibraryPath(file("../../libs/native/libraries/build/platform/").toString())
-  jvmArgs '-Xms4g', '-Xmx4g', '-Djava.util.concurrent.ForkJoinPool.common.parallelism=8', '-XX:+UnlockDiagnosticVMOptions', '-XX:+DebugNonSafepoints', '-XX:+HeapDumpOnOutOfMemoryError'
+  jvmArgs '-Xms16g', '-Xmx16g', '-Djava.util.concurrent.ForkJoinPool.common.parallelism=8', '-XX:+UnlockDiagnosticVMOptions', '-XX:+DebugNonSafepoints', '-XX:+HeapDumpOnOutOfMemoryError'
   if (buildParams.getRuntimeJavaVersion().map { it.majorVersion.toInteger() }.get() >= 21) {
     jvmArgs '--add-modules=jdk.incubator.vector', '--enable-native-access=ALL-UNNAMED'
   }
   if (System.getenv("DO_PROFILING") != null) {
     jvmArgs '-XX:StartFlightRecording=dumponexit=true,maxsize=250M,filename=knn.jfr,settings=profile.jfc'
   }
+  def asyncProfilerPath = System.getProperty("asyncProfiler.path", null)
+  if (asyncProfilerPath != null) {
+    if (OS.current().equals(OS.MAC)) {
+      def asyncProfilerAgent = "${asyncProfilerPath}/lib/libasyncProfiler.dylib"
+      println "Using async-profiler agent ${asyncProfilerAgent}"
+      jvmArgs "-agentpath:${asyncProfilerAgent}=start,event=cpu,interval=10ms,file=${layout.buildDirectory.asFile.get()}/tmp/elasticsearch-0_%t_%p.jfr"
+    } else if (OS.current().equals(OS.LINUX)) {
+      def asyncProfilerAgent = "${asyncProfilerPath}/lib/libasyncProfiler.so"
+      println "Using async-profiler agent ${asyncProfilerAgent}"
+      jvmArgs "-agentpath:${asyncProfilerAgent}=start,event=cpu,interval=10ms,wall=50ms,file=${layout.buildDirectory.asFile.get()}/tmp/elasticsearch-0_%t_%p.jfr"
+    } else {
+      println "Ignoring 'asyncProfiler.path': not available on ${OS.current()}";
+    }
+  }
```

Comment:
I am cool with this. However, why don't we add wall to MAC as well?

Reply:
I tried it and had to back off. Looking at the error I got and at the async-profiler code, apparently only Linux has an implementation that uses perf events, which let you record both cpu time and wall time at the same time. On Mac, the engine behind is less flexible/precise, and you can have one or the other.

I'm wondering: maybe I should add an option of that, like adding a `-DasyncProfiler.event=`, default to `cpu`, but can be changed to `wall` so Mac users can have this choice?